lib/ansible_tower_client/base_models/workflow_job_node.rb

```diff
@@ -5,7 +5,7 @@ def workflow_job
     end
 
     def job
-      job_id.nil? ? nil : api.jobs.find(job_id)
+      super.nil? ? nil : api.jobs.find(job_id)
```

Comment:
@jameswnl - Any way to get some specs around this change?

Reply:
This is a generic problem with the way that the BaseModel works, right?

Reply:
right. `job_id` doesn't exist if `job/super` is not there.

Reply:
@gmcculloug @syncrou any help on how to spec on this would be greatly appreciated

Reply:
@jameswnl - I was able to write a test supporting the `super.nil?` case:
```
  it "#job returns a nil when the job_id is nil" do
    obj = api.workflow_job_nodes.all.first
    allow_any_instance_of(AnsibleTowerClient::BaseModel).to receive(:job).and_return(nil)
    obj.job_id = 1
    expect(obj.job).to be_nil
  end
```
Though, still forced to include `allow_any_instance_of` which is problematic.

I'm inclined to approve this as is, with my work on writing the tests I'm confident it will work as expected.